WordPress/Classes/ViewRelated/Jetpack/Social/JetpackSocialNoConnectionView.swift

```diff
@@ -59,20 +59,81 @@ extension JetpackSocialNoConnectionView {
 
 // MARK: - View model
 
-struct JetpackSocialNoConnectionViewModel {
+class JetpackSocialNoConnectionViewModel: ObservableObject {
     let padding: EdgeInsets
     let hideNotNow: Bool
     let onConnectTap: (() -> Void)?
     let onNotNowTap: (() -> Void)?
+    @MainActor @Published var icons: [UIImage] = [UIImage()]
 
-    init(padding: EdgeInsets = Constants.defaultPadding,
+    init(services: [PublicizeService] = [],
+         padding: EdgeInsets = Constants.defaultPadding,
          hideNotNow: Bool = false,
          onConnectTap: (() -> Void)? = nil,
          onNotNowTap: (() -> Void)? = nil) {
         self.padding = padding
         self.hideNotNow = hideNotNow
         self.onConnectTap = onConnectTap
         self.onNotNowTap = onNotNowTap
+        updateIcons(services)
+    }
+
+    enum JetpackSocialService: String {
+        case facebook
+        case twitter
+        case tumblr
+        case linkedin
+        case unknown
+
+        var image: UIImage? {
+            switch self {
+            case .facebook:
+                return UIImage(named: "icon-facebook")
+            case .twitter:
+                return UIImage(named: "icon-twitter")
+            case .tumblr:
```

Comment:
Just for Tumblr, we'll need to differentiate the icon between light and dark schemes. Same with my previous comment, I'll let you decide whether to add them here or do it later.

Reply:
Is there a reason we need to separate the light and dark versions? It's currently grouped with a light and dark icon and automatically uses the correct one:

![Screenshot 2023-06-26 at 6 22 03 PM](https://github.com/wordpress-mobile/WordPress-iOS/assets/2454408/52fc178e-10e6-444c-8cbc-0792d5d7d349)

I did need to update the dark icon asset though. I did that in https://github.com/wordpress-mobile/WordPress-iOS/pull/20823/commits/7ebf53681da209855c3f3c099f2af6c79a8c516c.

Reply:
> Is there a reason we need to separate the light and dark versions? It's currently grouped with a light and dark icon and automatically uses the correct one

Yes, separating light and dark from the assets works for me. 👍🏼